osm2gtfs/creators/stops_creator.py

```diff
@@ -18,85 +15,99 @@ def __repr__(self):
         return rep
 
     def add_stops_to_feed(self, feed, data):
-        # Get stops information
-        stops = data.get_stops()
-
-        # add all stops to GTFS
-        for elem in stops.values():
-            if type(elem) is StopArea:
-                if len(elem.stop_members) > 1:
-                    parent_station = self.add_stop(feed, elem, None, True)
-                    for stop in elem.stop_members.values():
-                        self.add_stop(feed, stop, parent_station)
-                else:
-                    stop = elem.stop_members.values()[0]
-                    self.add_stop(feed, stop)
-            else:
-                self.add_stop(feed, elem)
-
-        # Add loose stop objects to route objects
-        self.add_stops_to_routes(data)
-
-    def add_stop(self, feed, stop, parent_station=None, is_station=False):
-        stop_dict = {"stop_lat": float(stop.lat),
-                     "stop_lon": float(stop.lon),
-                     "stop_name": stop.name}
-
-        if is_station:
-            stop_dict["stop_id"] = "SA" + str(stop.id)
-            stop_dict["location_type"] = "1"
-        else:
-            stop_dict["stop_id"] = str(stop.id)
-            stop_dict["location_type"] = ""
+        """
+        This function adds the Stops from the data to the GTFS feed.
+        It also unites stops in stations based on stop_areas in OpenStreetMap.
+        """
+        all_stops = data.get_stops()
+        regular_stops = all_stops['regular']
+        parent_stations = all_stops['stations']
 
-        if parent_station is None:
-            stop_dict["parent_station"] = ""
-        else:
-            stop_dict["parent_station"] = parent_station.stop_id
+        # Loop through all stations and prepare stops
+        for station in parent_stations.values():
+
+            # Add station to feed as stop
+            gtfs_stop_id = self._add_stop_to_feed(station, feed)
+
+            # Loop through member stops of a station
+            for member in station.get_members():
+
+                # Set parent station of each member Stop
+                regular_stops[member].set_parent_station(gtfs_stop_id)
+
+        # Loop through regular stops
+        for stop in regular_stops.values():
+
+            # Add stop to feed
+            gtfs_stop_id = self._add_stop_to_feed(stop, feed)
+
+    def _add_stop_to_feed(self, stop, feed):
+        """
+        This function adds a single Stop or Station object as a stop to GTFS.
+        It can be overridden by custom creators to change how stop_ids are made
+        up.
+
+        :return stop_id: A string with the stop_id in the GTFS
+        """
+        try:
+            parent_station = stop.get_parent_station()
+        except AttributeError:
+            parent_station = ""
```

Comment:
Why do you use an empty string here instead of `None`?

Reply:
Because I expected GTFS to use an empty string here. But `None` seems to work just as good, and is better to read. Changed accordingly.